src/raphson.jl

```diff
@@ -172,6 +182,43 @@ function perform_step!(cache::NewtonRaphsonCache{false})
     return nothing
 end
 
+function objective_linesearch(cache::NewtonRaphsonCache) ## returns the objective functions required in LS
```

Comment:
This function `objective_linesearch` returns 3 functions and is called in a loop (so it might get called multiple times). is this an efficient approach?